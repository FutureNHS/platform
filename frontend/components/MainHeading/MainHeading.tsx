import React from "react";

import styled from "styled-components";

const StyledHorizontalRule = styled.hr`
  margin-bottom: 25px;
  ${({ theme }) => `
    color: ${theme.colorNhsukGrey1};
    border-bottom: 1px solid ${theme.colorNhsukGrey1};
  `}
`;

interface Props {
  children: string;
  withBorder?: boolean;
}

/**
 * This component should be used on every page as the main heading.
 * It's a workaround for a known next.js accessibility issue that doesn't read out
 * headlines on navigation
 */
const MainHeading = ({ children, withBorder }: Props) => {
  return (
    <>
      <h1 aria-live="polite">{children}</h1>
      {withBorder && <StyledHorizontalRule />}
    </>
  );
};

export default MainHeading;
