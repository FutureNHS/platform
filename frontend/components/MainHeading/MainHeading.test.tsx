import React from "react";

import { render } from "@testing-library/react";

import MainHeading from "./MainHeading";

test("takes a snapshot of the component", () => {
  const children = "example heading";
  const { asFragment } = render(<MainHeading>{children}</MainHeading>);

  expect(asFragment()).toMatchSnapshot();
});

test("takes a snapshot with the prop", () => {
  const children = "example heading";
  const { asFragment } = render(
    <MainHeading withBorder>{children}</MainHeading>
  );

  expect(asFragment()).toMatchSnapshot();
});
