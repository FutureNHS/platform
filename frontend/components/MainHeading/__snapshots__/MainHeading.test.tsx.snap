// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`takes a snapshot of the component 1`] = `
<DocumentFragment>
  <h1
    aria-live="polite"
  >
    example heading
  </h1>
</DocumentFragment>
`;

exports[`takes a snapshot with the prop 1`] = `
<DocumentFragment>
  <h1
    aria-live="polite"
  >
    example heading
  </h1>
  .c0 {
  margin-bottom: 25px;
  color: undefined;
  border-bottom: 1px solid undefined;
}

<hr
    class="c0"
  />
</DocumentFragment>
`;
